src/core/tsi/transport_security.h

```diff
@@ -111,6 +111,9 @@ tsi_result tsi_construct_allocated_string_peer_property(
 tsi_result tsi_construct_string_peer_property_from_cstring(
     const char *name, const char *value, tsi_peer_property *property);
 
+/* Destroy plugin registration funcion. */
+void tsi_register_destroy_plugin(void (*destroy)(void));
```

Comment:
I don't think the dependency issue is a problem, since any given TSI implementation would only be used if it is actually registered, in which case we will depend on it anyway.

Stepping back a bit, I think there are two basic choices here:

1. We use the existing gRPC plugin mechanism for TSI implementations.
2. We add a new registration mechanism for TSI implementations.

As I mentioned earlier, we do have a number of cases where we have gone with option 2, such as our existing resolver and LB policy APIs.  However, the reason that we added a new mechanism in those cases is that they are designed to register an implementation by name such that it can be selected by name at run-time, so there is factory machinery involved.  I don't think that's the case for the TSI implementation -- correct me if I'm wrong, but my understanding is that the TSI implementation is essentially chosen by the credentials, which are passed into gRPC via the API.  In that case, all we really need for TSI implementations is a hook to do global initialization and destruction, so I think the gRPC plugin mechanism will work fine.

If we do support a separate registration mechanism for TSI implementations, then the idea would be that the application would need to call `tsi_register_implementation()` before calling `grpc_init()`.  Then `grpc_init()` would call the init function of every TSI implementation that was registered, and `grpc_shutdown()` would call the destroy function of every TSI implementation that was registered.  But as I said above, this is functionally equivalent to just calling the init and destroy functions as a gRPC plugin.

Given the above, I think we should just use the existing gRPC plugin mechanism instead of creating a new one.